sky/backends/cloud_vm_ray_backend.py

```diff
@@ -265,6 +265,25 @@ def add_prologue(self, job_id: int, is_local: bool = False) -> None:
                 log_to_driver=True,
                 **kwargs
             )
+            def wait_or_fail(futures) -> List[int]:
+                \"\"\"Wait for tasks, if any fails, cancel all unready.\"\"\"
+                returncodes = []
+                # Wait for at leat 1 task to be ready.
+                ready, unready = ray.wait(futures)
+                while unready:
```

Comment:
If all tasks finish immediately, we won't enter this loop and `returncodes` won't be assigned?